tests/Makefile

```diff
@@ -262,22 +262,22 @@ test-lz4-hugefile: lz4 datagen
 
 test-lz4-testmode: lz4 datagen
 	@echo "\n ---- bench mode ----"
-	$(LZ4) -bi1
+	  $(LZ4) -bi1
 	@echo "\n ---- test mode ----"
-	./datagen | $(LZ4) -t             && false || true
-	./datagen | $(LZ4) -tf            && false || true
+	! ./datagen | $(LZ4) -t
```

Comment:
"|" binds more tightly than "!", so this does the right thing here.

According to [POSIX](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_09_02):

> If the pipeline [i.e. series of commands separated by `|`] does not begin with the ! reserved word, the exit status shall be the exit status of the last command specified in the pipeline. Otherwise, the exit status shall be the logical NOT of the exit status of the last command.

&lt;aside&gt;

To get the behaviour you _don't_ want here, but which is sometimes useful[1], there's no standard way that I know of, but `bash` has a `pipefail` extension; if that option is set, the exit status of the pipeline as a whole is the last _nonzero_ status of any of the simple commands (or 0 if they all succeed).  But even there, "!" inverts the exit status of the pipeline as a whole.  (And you're not setting `pipefail` in any case...)

[1] Here's a case where it's the first command whose status one wants:

     make test | tee logfile || echo "test failed"

(which (minus the "|| echo" part) is how I got here in the first place -- trying to capture the output of "make test" triggered #308 :-/ ).

&lt;/aside&gt;